framework/helpers/BaseHtml.php

```diff
@@ -236,7 +236,7 @@ public static function cssFile($url, $options = [])
         if (!isset($options['rel'])) {
             $options['rel'] = 'stylesheet';
         }
-        $options['href'] = Url::to($url);
+        $options['href'] = Yii::$app->getUrlManager()->to($url);
```

Comment:
While moving logic from helper to class is a good change, I think methods should be renamed because now they don't describe what are they doing indeed.

_"Url manager to $url"_ &ndash; whaat?

When it was `Url::to()` it was pretty clear what does the `to()` method does. Now it's not like that :) I don't have a good idea now and discussion is opened.

Reply:
```
Yii::$app->getUrlManager()->urlTo($url)
Yii::$app->urlManager->urlTo($url)
$um = Yii::$app->urlManager;
$um->urlTo($url)
```

| Old | New |
|-----------------|-----|
| `Url::toRoute($route, false)` | `Yii::$app->urlManager->urlToRoute($route, false)`  |
| `Url::normalizeRoute($route)` | `Yii::$app->urlManager->normalizeRoute($route)`  |
| `Url::to('', false)` | `Yii::$app->urlManager->to('', false)`  |
| `Url::ensureScheme($url, $scheme)` | `Yii::$app->urlManager->ensureUrlScheme($url, $scheme)`  |
| `Url::base(false)` | `Yii::$app->urlManager->baseUrl(false)`  |
| `Url::remember('', null)` | `Yii::$app->urlManager->rememberUrl('', null)`  |
| `Url::previous(null)` | `Yii::$app->urlManager->previousUrl(null)`  |
| `Url::canonical()` | `Yii::$app->urlManager->canonicalUrl()`  |
| `Url::home(false)` | `Yii::$app->urlManager->homeUrl(false)`  |
| `Url::isRelative($url)` | `Yii::$app->urlManager->isRelativeUrl($url)`  |
| `Url::current([], false)` | `Yii::$app->urlManager->currentUrl([], false)`  |

How about using getter syntax for methods that are usually used without arguments?

| Old | New |
|-----------------|-----|
| `Url::base(false)` | `Yii::$app->urlManager->getBaseUrl(false)`  |
| `Url::base(false)` | `Yii::$app->urlManager->baseUrl`  |
| `Url::previous(null)` | `Yii::$app->urlManager->getPreviousUrl(null)`  |
| `Url::previous(null)` | `Yii::$app->urlManager->previousUrl`  |
| `Url::canonical()` | `Yii::$app->urlManager->getCanonicalUrl()`  |
| `Url::canonical()` | `Yii::$app->urlManager->canonicalUrl`  |
| `Url::home(false)` | `Yii::$app->urlManager->getHomeUrl(false)`  |
| `Url::home(false)` | `Yii::$app->urlManager->homeUrl`  |
| `Url::current([], false)` | `Yii::$app->urlManager->getCurrentUrl([], false)`  |
| `Url::current([], false)` | `Yii::$app->urlManager->currentUrl`  |

Reply:
@SilverFire,  how about this renaming methods?

| old | new |
|---|---|
| `to()` | `createUrlTo()` |
| `toRoute()` | `createUrlToRoute()` |
| `base($scheme = false)` + `getBaseUrl()` | `getBaseUrl($scheme = false)` |
| `home($scheme = false)` + `getHomeUrl()` | `getHomeUrl($scheme = false)` |